cf/field.py

```diff
@@ -16062,16 +16068,22 @@ def regrids(
                 config={"coord": x, "period": Data(360.0, "degrees")},
             )
 
-        # Release old memory from ESMF (this ought to happen garbage
-        # collection, but it doesn't seem to work there!)
-        if destroy_old_Regrid:
-            regridSrc2Dst.destroy()
+        # Explicitly release all the memory that will not be needed anymore
+        if not dst_regrid:
+            # explicitly release memory for destination ESMF objects
+            # as they were only created locally (i.e. not originating
+            # from an existing regrid operator) and they will not be
+            # used anymore
             dstfracfield.destroy()
-            srcfracfield.destroy()
             dstfield.destroy()
-            srcfield.destroy()
             dstgrid.destroy()
-            srcgrid.destroy()
+
+            if not return_operator:
```

Comment:
I think that `del regridSrc2Dst` can be run every time, sine  if `return_operator` were `True` we would already have returned by now.

Reply:
Absolutely. Fixed.